pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -451,15 +451,27 @@ func monitorFor(p *Propagator, clientPool dynamic.ClientPool, resource unversion
 				// namespaces if it's namespace scoped, so leave
 				// APIResource.Namespaced as false is all right.
 				apiResource := unversioned.APIResource{Name: resource.Resource}
-				return client.Resource(&apiResource, api.NamespaceAll).List(&options)
+				// The parameter codec used by the dynamic client cannot encode
```

Comment:
> If we specified a parameter codec when we created the dynamic client, would we have this problem?

Do you mean make the parameter codec a member of the dynamic client and pass it in when creating the client? That would work.

> Also, the dynamic client should probably fail if you pass it internal objects and it can't version them.

I don't think the dynamic client should do conversion. The whole idea of the dynamic client is that it doesn't need to know about the scheme. I think the caller is responsible to pass in a struct with tags indicating how to convert the struct to queries.